### DISCUSSION TITLE
Configuration

### DISCUSSION THREAD

**Discussion by thboss** [PhlexPlexico/G5API#107] [Help] [answered]
Hey @PhlexPlexico, I'm trying to set up the API, and I have a domain but I'm confused about what I should set in these fields `hostname` `clientHome` `apiURL`

I don't want to set up G5V since I just use the API for a discord bot.

**Comment by PhlexPlexico** [comment, marked as answer]
> The clientHome is a URL to use on redirects during authentication. This is what your front-end URL is supposed to be. In this case, G5V uses localhost:8080 to run its development server on, so it is set to that. If you wish to use this over the network, and not just your machine, set it to your front-ends local IP address. CORS is also enabled for that URL specifically, to prevent any unauthorized usage of the API.

Since you're just using the API to login and redirect back to it, you can just set it to the API URL - in this case if it's running on port 3300, it would be `http://yourhost.me:3300`.

> The apiURL portion in the config is the URL that is going to be sent to the counter-strike servers. Please set this as a direct link to your API. Be it through a reverse proxy (via nginx, webpack, etc), or a direct link to the API, where the welcome page is displayed (i.e. message: "Welcome to G5API!" upon login).

So that would now be the same as `clientHome` :)

> Next is the hostname. Since get5 requires a valid hostname, you can not simply use an IP. Please register a domain, or use something like DuckDNS to create a hostname!

So hostname would just be without the port - `http://yourhost.me`

**Reply by thboss** [reply]
Thanks for quick response,

Do I need to set Apache/Nginx configuration for that hostname?

**Reply by thboss** [reply]
I get it works by set all ports to 3301

**Reply by PhlexPlexico** [reply]
Then there you go! It doesn't need nginx since you're not running a reverse proxy and just connecting directly
